File: seastarvideo/src/main/java/com/freewind/seastarvideo/api/ApiImpl.java
package com.freewind.seastarvideo.api;

import com.freewind.seastarvideo.bean.BaseBean;

public interface ApiImpl<T extends BaseBean> {
    void onSuccess(T data);

    void onFailed(int code, String msg);

    void onStart();

    void onCancel();
}

File: seastarvideo/src/main/java/com/freewind/seastarvideo/http/ApiImpl.kt
package com.freewind.seastarvideo.http

import com.freewind.seastarvideo.http.bean.BaseBean

interface ApiImpl<T : BaseBean> {
    fun onSuccess(data: T)

    fun onFailed(code: Int, msg: String?)

    fun onStart()

    fun onCancel()
}
